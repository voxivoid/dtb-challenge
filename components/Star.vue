<template lang="pug">
div
  i.fas.fa-star(v-if="starred")
  i.far.fa-star(v-else)
</template>

<script>
export default {
  props: {
    starred: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="stylus" scoped>

</style>
