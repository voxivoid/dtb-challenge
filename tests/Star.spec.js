import { mount } from "@vue/test-utils";

import Star from "@/components/Star.vue";

describe("Star", () => {
  const wrapper = mount(Star);

  it("shows the correct icon when not starred", () => {
    wrapper.setProps({ starred: false });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("shows the correct icon when starred", () => {
    wrapper.setProps({ starred: true });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
