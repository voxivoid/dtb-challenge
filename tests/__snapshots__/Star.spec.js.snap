// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star shows the correct icon when not starred 1`] = `<div class="star"><i class="el-icon-star-off"></i></div>`;

exports[`Star shows the correct icon when starred 1`] = `<div class="star"><i class="el-icon-star-on"></i></div>`;
